dependency/test/test.sh

```diff
@@ -0,0 +1,83 @@
+
+
+#!/bin/bash
+
+set -euo pipefail
+
+extract_tarball() {
+  rm -rf icu
+  mkdir icu
+  tar --extract \
+    --file "$1" \
+    --directory icu
+}
+
+set_ld_library_path() {
+  export LD_LIBRARY_PATH="$PWD/icu/lib:${LD_LIBRARY_PATH:-}"
+}
+
+check_version() {
+  expected_version=$1
+  actual_version="$(/icu/bin/icuinfo 2> /dev/null | sed -rn 's/.*param name="version">([0-9\.]+).*/\1/p')"
+  if [[ "${actual_version}" != "${expected_version}" ]]; then
+    echo "Version ${actual_version} does not match expected version ${expected_version}"
+    exit 1
+  fi
+}
+
+check_file() {
+  if ! test -f icu/lib/libicudata.so; then
+    echo "Library file missing"
+    exit 1
+  fi
+}
+
+main() {
```

Comment:
I notice that this test will run on the local runner. In many other dependencies we run the test against the environment that the dependency will run in. Here, since we're compiling ICU once for both Jammy and Bionic, it probably makes sense to have the test script run the tests in both a Jammy and Bionic docker image environment.

Check out how this is done in bundler, because those tests do a similar thing:

https://github.com/paketo-buildpacks/bundler/blob/f39310c2f99246e91248731e8399145a3bf60c6f/dependency/test/run-test#L39-L45